### PR TITLE
iOS v4 solution switching

### PR DESCRIPTION
**Pull request by ChristianWolf-mapspeople**

### CODE REVIEW

File: docs/06-Working with Data/04-Various/switch-solutions.mdx [hunk 1]
```diff
@@ -147,7 +147,64 @@ fun initMapControl(view: View) {
...
+<TabItem value="ios-v4" label="iOS v4">
+
+When you load your initial Solution, it's beneficial to initialise MapsIndoors properly, to ensure it's easy to switch Solutions later if needed.
+
+<Tabs groupId="various-switch-solutions-platform">
+
+<TabItem value="googlemaps" label="Google Maps">
+
+```swift
+func setupMapIndoors(mapsIndoorsAPIKey: String, googleMapsAPIKey: String) async throws {
```
Comment (fhansenMP) [inline docs/06-Working with Data/04-Various/switch-solutions.mdx:159]: setupMap**s**Indoors

File: docs/06-Working with Data/04-Various/switch-solutions.mdx [hunk 2]
```diff
@@ -147,7 +147,64 @@ fun initMapControl(view: View) {
 </Tabs>
 
 </TabItem>
-<TabItem value="ios" label="iOS v3">
+<TabItem value="ios-v4" label="iOS v4">
+
+When you load your initial Solution, it's beneficial to initialise MapsIndoors properly, to ensure it's easy to switch Solutions later if needed.
```
Comment (fhansenMP) [inline docs/06-Working with Data/04-Various/switch-solutions.mdx:152]: initiali**z**e

File: docs/06-Working with Data/04-Various/switch-solutions.mdx [hunk 3]
```diff
@@ -147,7 +147,64 @@ fun initMapControl(view: View) {
...
+    self.mapControl = MPMapsIndoors.createMapControl(mapConfig: mapConfig)
+}
+```
+
+</TabItem>
+
+<TabItem value="mapbox" label="Mapbox">
+
+```swift
+func setupMapIndoors(mapsIndoorsAPIKey: String, mapboxAPIKey: String) async throws {
```
Comment (fhansenMP) [inline docs/06-Working with Data/04-Various/switch-solutions.mdx:181]: same as previous comment

Comment (fhansenMP) [inline docs/06-Working with Data/04-Various/switch-solutions.mdx:181]: I know it's functionally the same - but mapbox likes to refer to their api key as an _access token_. Should we respect that, or just generalize? I'm not sure myself

Comment (ChristianWolf-mapspeople) [reply docs/06-Working with Data/04-Various/switch-solutions.mdx:181]: Let's do that :)

File: docs/06-Working with Data/04-Various/switch-solutions.mdx [hunk 4]
```diff
@@ -250,7 +307,43 @@ initMapControl(mMapBoxMap, mMapView)
 </Tabs>
 
 </TabItem>
-<TabItem value="ios" label="iOS v3">
+<TabItem value="ios-v4" label="iOS v4">
+
+To switch Solutions, you first need to ensure that all your existing instances are closed down safely. Once this is done, you can use `setupMapIndoors` to restart with your new desired API key.
```
Comment (fhansenMP) [inline docs/06-Working with Data/04-Various/switch-solutions.mdx:312]: It's good practice but not required - it's perfectly fine to invoke "load()" without explicitly shutting down a previous instance. We handle that internally anyways 😄

Comment (ChristianWolf-mapspeople) [reply docs/06-Working with Data/04-Various/switch-solutions.mdx:312]: Let's make the description as simple as possible 😄

**Comment by ChristianWolf-mapspeople** [comment]
> It's also important to note that people shouldn't try to re-use MapControl instances. When switching solutions, any previous MapControl is invalid and has undefined behavior - so it's required to instantiate a new one.

This is what happens in `setupMapsIndoors()`

**Review by fhansenMP — APPROVED** [review]

**Pull request merged**